src/libraries/Microsoft.PowerFx.Core/Functions/Delegation/DelegationStrategies/DelegationValidationStrategy.cs

```diff
@@ -344,84 +403,51 @@ public bool IsValidCallNode(CallNode node, TexlBinding binding, OperationCapabil
-                var nodeType = binding.GetType(node);
-
-                if (leftType != null && nodeType != null)
-                {
-                    if ((leftType.Kind == DKind.OptionSet && nodeType.Kind == DKind.OptionSetValue) || (leftType.Kind == DKind.View && nodeType.Kind == DKind.ViewValue))
-                    {
-                        // OptionSet and View Access are delegable despite being async
-                        return true;
-                    }
-                }
-                else
-                {
-                    return false;
-                }
+                return true;
             }
 
-            if (node is CallNode callNode)
+            // Async predicates and impure nodes are not supported unless Features say otherwise.
+            // Let CallNodes for delegatable async functions be marked as being Valid to allow
+            // expressions with delegatable async function calls to be delegated
+
+            // Impure nodes should only be marked valid when Feature is enabled.
+            if (!isPure && !binding.Features.HasFlag(Features.AllowImpureNodeDelegation))
             {
-                var callInfo = binding.GetInfo(callNode);
-                if (callInfo != null)
-                {
-                    if (binding.IsBlockScopedConstant(node) || (callInfo.Function is AsTypeFunction))
-                    {
-                        // AsType is delegable despite being async
-                        return true;
-                    }
-                }
-                else
-                {
-                    return false;
-                }
+                TrackingProvider.Instance.SetDelegationTrackerStatus(DelegationStatus.ImpureNode, node, binding, Function, DelegationTelemetryInfo.CreateImpureNodeTelemetryInfo(node, binding));
             }
-
-            // Async predicates and impure nodes are not supported.
-            // Let CallNodes for User() be marked as being Valid to allow
-            // expressions with User() calls to be delegated
-            if (!IsUserCallNodeDelegable(node, binding) && (isAsync || !isPure))
+            else
             {
-                var telemetryMessage = string.Format(CultureInfo.InvariantCulture, "Kind:{0}, isAsync:{1}, isPure:{2}", node.Kind, isAsync, isPure);
-                SuggestDelegationHintAndAddTelemetryMessage(node, binding, telemetryMessage);
-
-                if (isAsync)
+                // If the feature is enabled, enable delegation for
+                // async call, first name and dotted name nodes.
+                if (binding.Features.HasFlag(Features.AllowAsyncDelegation))
                 {
-                    TrackingProvider.Instance.SetDelegationTrackerStatus(DelegationStatus.AsyncPredicate, node, binding, Function);
+                    return (node is CallNode) || (node is FirstNameNode) || (node is DottedNameNode);
                 }
```

Comment:
Do you need to add delegation hit for following scenario

isAync && !isPure 

In this case, currently you return true without checking the value of Features.AllowImpureNodeDelegation). Just want to confirm if this is the behavior that you want.